ambry-mysql/src/main/java/com/github/ambry/repair/MysqlRepairRequestsDb.java

```diff
@@ -0,0 +1,216 @@
+        statement.setBytes(1, Base64.decodeBase64(blobId));
+        statement.setShort(2, (short) operationType.ordinal());
+        statement.executeUpdate();
+      }
+    } catch (SQLException e) {
+      // LOCAL_CONSISTENCY_TODO: add metrics for all the SQLException.
+      logger.error("failed to delete record from {} due to {}", dataSource, e.getMessage());
+      throw e;
+    }
+  }
+
+  /**
+   * Insert one {@link RepairRequestRecord}
+   * @param record the record to insert
+   */
+  public void putRepairRequests(RepairRequestRecord record) throws SQLException {
+    // private static final String INSERT_QUERY = String.format(""
+    //    + "INSERT INTO %1$s "
+    //    + "(%2$s, %3$s, %4$s, %5$s, %6$s, %7$s, %8$s, %9$s) VALUES (?, ?, ?, ?, ?, ?, ?, ?)",
+    //    REPAIR_REQUESTS_TABLE,
+    //    BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE);
+    try (Connection connection = dataSource.getConnection()) {
+      try (PreparedStatement statement = connection.prepareStatement(INSERT_QUERY)) {
+        statement.setBytes(1, Base64.decodeBase64(record.getBlobId()));
+        statement.setLong(2, record.getPartitionId());
+        statement.setString(3, record.getSourceHostName());
+        statement.setInt(4, record.getSourceHostPort());
+        statement.setShort(5, (short) record.getOperationType().ordinal());
+        statement.setTimestamp(6, new Timestamp(record.getOperationTimeMs()));
+        statement.setShort(7, record.getLifeVersion());
+        if (record.getExpirationTimeMs() != Utils.Infinite_Time) {
+          statement.setTimestamp(8, new Timestamp(record.getExpirationTimeMs()));
+        } else {
+          statement.setTimestamp(8, null);
+        }
+        statement.executeUpdate();
+      }
+    } catch (SQLException e) {
+      logger.error("failed to insert record to {} due to {}", dataSource, e.getMessage());
+      throw e;
+    }
+  }
+
+  /**
+   * Select the records from one partition ordered by the operation time.
+   * @param partitionId partition id
+   * @return the oldest {@link RepairRequestRecord}s.
+   */
+  public List<RepairRequestRecord> getRepairRequests(long partitionId) throws SQLException {
+    // private static final String GET_QUERY = String.format(""
+    //    + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+    //    + "FROM %s "
+    //    + "WHERE %s = ? "
+    //    + "ORDER BY %s ASC "
+    //    + "LIMIT 100",
+    //    BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE,
+    //    REPAIR_REQUESTS_TABLE,
+    //    PARTITION_ID,
+    //    OPERATION_TIME);
+    ResultSet resultSet = null;
+    try (Connection connection = dataSource.getConnection()) {
+      try (PreparedStatement statement = connection.prepareStatement(GET_QUERY)) {
+        statement.setLong(1, partitionId);
+        resultSet = statement.executeQuery();
```

Comment:
we can use a try-close statement to get the ResultSet, so you don't have to call closeQuietly in the catch clause.

Reply:
Changed.